fastestimator/network.py

```diff
@@ -483,21 +500,20 @@ def transform(self, data: Dict[str, Any], mode: str, epoch: int = 1) -> Dict[str
         data = to_tensor(data, target_type="tf")
         data, prediction = self.run_step(data)
         self.unload_epoch()
+        batch_size = get_batch_size(data)
+        # handle tensorflow multi-gpu inferencing issue, it will replicate data on each device
+        if isinstance(tf.distribute.get_strategy(), tf.distribute.MirroredStrategy):
+            for key, val in prediction.items():
+                if hasattr(val, "shape") and list(val.shape) and val.shape[0] > batch_size:
```

Comment:
does this need to consider nested dict predictions?

Reply:
yes, in later PR. I already have this noted down.    the recursive operation needs  to be done for other function too .

Reply:
Add a # TODO

Reply:
done